Source/JavaScriptCore/wasm/js/WebAssemblyModuleRecord.cpp

```diff
@@ -616,15 +616,21 @@ void WebAssemblyModuleRecord::initializeExports(JSGlobalObject* globalObject)
 
             switch (global.bindingMode) {
             case Wasm::GlobalInformation::BindingMode::EmbeddedInInstance: {
-                m_instance->instance().setGlobal(globalIndex, initialBits);
+                if (Wasm::isRefType(global.type))
```

Comment:
What marks these bits?

Reply:
The Wasm Instance does, here: https://searchfox.org/wubkat/source/Source/JavaScriptCore/wasm/js/JSWebAssemblyInstance.cpp#97 for the embedded globals. For portable globals, it's done by the JS Wasm global object here: https://searchfox.org/wubkat/source/Source/JavaScriptCore/wasm/WasmGlobal.cpp#152

Looking at this closer though, I think the latter should be generalized to check `isRefType` instead of what it's checking now... I can make that change here too.

The reason why it was crashing before though was that it failed to go through the writebarrier via `setGlobal` https://searchfox.org/wubkat/source/Source/JavaScriptCore/wasm/WasmInstance.cpp#104

Reply:
Pushed a fix & test for the `WasmGlobal` marking issue